content/explugins/fallback.md

```diff
@@ -0,0 +1,83 @@
++++
+title = "fallback"
+description = "*fallback* allows redirecting queries to an alternate set of upstreams based on RCODE"
+weight = 10
+tags = [  "plugin" , "fallback" ]
+categories = [ "plugin", "external" ]
+date = "2018-02-07T12:36:00-05:00"
+repo = "https://github.com/coredns/fallback/"
+home = "https://github.com/coredns/fallback/blob/master/README.md"
++++
+
+# fallback
+
+## Name
+
+*fallback* - a plugin that sends queries to an alternate set of upstreams if the plugin
+chain returns specific error messages.
+
+## Description
+
+The *fallback* plugin allows an alternate set of upstreams be specified which will be used
+if the plugin chain returns specific error messages. The *fallback* plugin utilizes the *proxy* plugin (<https://coredns.io/plugins/proxy>) to query the specified upstreams.
+
+As the name suggests, the purpose of the *fallback* is to allow a fallback when, for example,
+the desired upstreams became unavailable.
+
+It is recommended that this comes before *proxy* and *forward* in the `plugins.cfg` file. It must
+come before any plugins for which it wants to react to their response code.
+
+## Syntax
+
+```
+{
+    fallback RCODE PROXY_PARAMS
+}
+```
+
+* **RCODE** is the string representation of the error response code. The complete list of valid rcode strings are defined as `RcodeToString` in <https://github.com/miekg/dns/blob/master/msg.go>, examples of which are `SERVFAIL`, `NXDOMAIN` and `REFUSED`.
+* **PROXY_PARAMS** accepts the same parameters as the *proxy* plugin
+<https://coredns.io/plugins/proxy>.
+
+## Examples
+
+### Fallback to local DNS server
+
+The following specifies that all requests are proxied to 8.8.8.8. If the response is `NXDOMAIN`, *fallback* will proxy the request to 192.168.1.1:53, and reply to client accordingly.
+
+```
```

Comment:
the stanza that are wrapped into 

~~~ corefile
...
~~~

are automatically tested into the test_readme.go
=> it runs a CoreDNS server with that stanza as corefile .. and check that the server can start without error.

A way to verify your stanza is ok .. or become wrong if there is a change in the plugin.

example:
```
~~~ corefile
. {
    erratic {
        drop 2
        truncate 2
    }
}
~~~
```

NOTE: the other stanze are wrapped into "~~~ txt ... ~~~"